test/e2e/framework/vault/provider.go

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package vault is helper functions for e2e
+package vault
+
+import (
+	"bufio"
+	"context"
+	"errors"
+	"io"
+	"net/http"
+
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/kubectl/pkg/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/secrets-store-csi-driver/test/e2e/framework"
+)
+
+const (
+	providerVaultYAML          = "https://raw.githubusercontent.com/hashicorp/secrets-store-csi-driver-provider-vault/master/deployment/provider-vault-installer.yaml"
+	providerVaultDaemonSetName = "csi-secrets-store-provider-vault"
+)
+
+type InstallProviderInput struct {
+	Creator   framework.Creator
+	Namespace string
+}
+
+func InstallProvider(ctx context.Context, input InstallProviderInput) {
```

Comment:
Sure. I tried to remove `kubectl` command dependencies as much as possible. however, we can't do that because we don't have such functions equivalent to `kubectl exec`. we still need `kubectl` for testing.